sdk/src/main/java/com/mapbox/maps/renderer/FpsManager.kt

```diff
@@ -0,0 +1,186 @@
+package com.mapbox.maps.renderer
+
+import android.os.Handler
+import androidx.annotation.WorkerThread
+import androidx.core.os.postDelayed
+import com.mapbox.maps.logI
+import com.mapbox.maps.logW
+import kotlin.math.pow
+
+@WorkerThread
+internal class FpsManager(
+  private val handler: Handler
+) {
+  private var userRefreshRate = USER_DEFINED_REFRESH_RATE_NOT_SET
+  private var userDefinedRatio = USER_DEFINED_RATIO_NOT_SET
+
+  private var screenRefreshRate = SCREEN_METRICS_NOT_DEFINED
+  private var screenRefreshPeriodNs = -1L
+  private var previousRefreshNs = -1L
+  private var previousDrawnFrameIndex = 0
+  private var frameRenderTimeAccumulatedNs = 0L
+
+  private var preRenderTimeNs = -1L
+  private var choreographerTicks = 0
+  private var choreographerSkips = 0
+
+  internal var fpsChangedListener: OnFpsChangedListener? = null
+
+  fun setScreenRefreshRate(screenRefreshRate: Int) {
+    if (this.screenRefreshRate == screenRefreshRate) {
+      return
+    }
+    this.screenRefreshRate = screenRefreshRate
+    screenRefreshPeriodNs = ONE_SECOND_NS / screenRefreshRate
+    if (userRefreshRate != USER_DEFINED_REFRESH_RATE_NOT_SET) {
+      userDefinedRatio = userRefreshRate.toDouble() / screenRefreshRate
+      logI(TAG, "User defined ratio is $userDefinedRatio")
+    }
+    if (LOG_STATISTICS) {
+      logI(
+        TAG,
+        "Device screen frequency is $screenRefreshRate," +
+          " user defined refresh rate is ${if (userRefreshRate == USER_DEFINED_REFRESH_RATE_NOT_SET) "not set" else userRefreshRate}"
+      )
+    }
+  }
+
+  fun setUserRefreshRate(refreshRate: Int) {
```

Comment:
is it possible to set `refreshRate` to 0?
maybe check it?

Reply:
we have a check for zero before we get here